caiman_napari/algorithms/cnmfe.py

```diff
@@ -0,0 +1,214 @@
+import pathlib
+import numpy as np
+import caiman as cm
+from caiman.source_extraction.cnmf import cnmf as cnmf
+from caiman.utils.utils import load_dict_from_hdf5
+from caiman.source_extraction.cnmf.params import CNMFParams
+from caiman.motion_correction import MotionCorrect
+from caiman.utils.utils import download_demo
+import psutil
+import json
+from tqdm import tqdm
+import sys
+import pandas as pd
+import pickle
+from caiman.utils.visualization import get_contours as caiman_get_contours
+from caiman.utils.visualization import inspect_correlation_pnr, nb_inspect_correlation_pnr
+from caiman.source_extraction.cnmf.cnmf import load_CNMF
+from caiman_napari.utils import *
+from caiman.summary_images import local_correlations_movie_offline
+import os
+import traceback
+from napari.viewer import Viewer
+import time
+from time import sleep
+
+def main(batch_path, uuid):
+    df = pd.read_pickle(batch_path)
+    item = df[df['uuid'] == uuid].squeeze()
+
+    input_movie_path = item['input_movie_path']
+    params = item['params']
+    print("cnmfe params:", params)
+
+    #adapted from current demo notebook
+    n_processes = psutil.cpu_count() - 1
+    # Start cluster for parallel processing
+    c, dview, n_processes = cm.cluster.setup_cluster(
+        backend='local',
+        n_processes=n_processes,
+        single_thread=False
+     )
+
+    try:
+        fname_new = cm.save_memmap(
+            [input_movie_path],
+            base_name='memmap_',
+            order='C',
+            dview=dview
+        )
+
+        print('making memmap')
+        gSig = params['cnmfe_kwargs']['gSig'][0]
+
+        Yr, dims, T = cm.load_memmap(fname_new)
+        images = np.reshape(Yr.T, [T] + list(dims), order='F')
+        downsample_ratio = params['downsample_ratio']
+        # in fname new load in memmap order C
+
+        cn_filter, pnr = cm.summary_images.correlation_pnr(
+            images[::downsample_ratio], swap_dim=False, gSig=gSig
```

Comment:
ah so downsample is generally just used for the corr pnr stuff?

Reply:
As far as I can tell, yes. Just to reduce processing time (as it doesn't affect the output according to the CNMFE demo).

Reply:
I see, I was just curious since I never used it.